Orange/preprocess/score.py

```diff
@@ -196,3 +203,70 @@ class Gini(ClassificationScorer):
     """
     def from_contingency(self, cont, nan_adjustment):
         return (_gini(np.sum(cont, axis=1)) - _gini(cont)) * nan_adjustment
+
+
+class ReliefF(Scorer):
+    feature_type = Variable
+    class_type = DiscreteVariable
+
+    def __init__(self, n_iterations=None, k_nearest=None):
+        self.n_iterations = n_iterations
+        self.k_nearest = k_nearest
+
+    def score_data(self, data, feature):
+        assert len(data.domain.class_vars) == 1, 'TODO: multi-class?'
+        assert data.domain.class_var.is_discrete
```

Comment:
It is better to raise an appropriate error (with an informative message) if conditions for the method are not met.